InfluxDB get started (#4628)
influxdata/docs-v2: content/influxdb/v2.4/get-started/setup.md

{{< api-endpoint endpoint="http://localhost:8086/api/v2/authorizations" method="post" >}}

Include the following with your request:

- **Headers**:
  - **Authorization**: Token <INFLUX_OPERATOR_TOKEN>
  - **Content-Type**: application/json
- **Request body**: JSON body with the following properties:
  - **status**: "active" or "inactive"
  - **description**: API token description
  - **orgID**: [InfluxDB organization ID](/influxdb/v2.4/organizations/view-orgs/#view-your-organization-id)
  - **permissions**: Array of objects where each object represents permissions
    for an InfluxDB resource. Each permission contains the following properties:
      - **action**: "read" or "write"
      - **resource**: JSON object that represents the InfluxDB resource to grant
        permission to. Each resource contains at least the following properties:
          - **orgID**: [InfluxDB organization ID](/influxdb/v2.4/organizations/view-orgs/#view-your-organization-id)
          - **type**: Resource type.
            _For information about what InfluxDB resource types exist, use the
            [`/api/v2/resources` endpoint](/influxdb/v2.4/api/#operation/GetResources)._

> 

Reviewer: The following example shows how to use cURL with the InfluxDB API to generate an all access token:

{{% truncate %}}
```sh
export INFLUX_HOST=http://localhost:8086
export INFLUX_ORG_ID=<YOUR_INFLUXDB_ORG_ID>

> export INFLUX_TOKEN=<YOUR_INFLUXDB_OPERATOR_TOKEN>

Reviewer: These are already set for OSS by the setup process.

      {"action": "read", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "checks"}},
      {"action": "write", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "checks"}},
      {"action": "read", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "dbrp"}},
      {"action": "write", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "dbrp"}},
      {"action": "read", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "notebooks"}},
      {"action": "write", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "notebooks"}},
      {"action": "read", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "annotations"}},
      {"action": "write", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "annotations"}},
      {"action": "read", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "remotes"}},
      {"action": "write", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "remotes"}},
      {"action": "read", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "replications"}},
      {"action": "write", "resource": {"orgID": "'"$INFLUX_ORG_ID"'", "type": "replications"}}
    ]
  }
'
```
{{% /truncate %}}

> The request returns a JSON object with the following properties:

Reviewer: ```suggestion
The response body contains a JSON object with the following properties:
```

- **id**: API Token ID
- **token**: API Token ({{< req "Important" >}})
- **status**: Token status
- **description**: Token description
- **orgID**: InfluxDB organization ID the token is associated with
- **org**: InfluxDB organization name the token is associated with
- **userID**: User ID the token is associated with

> - **user**: User ID the token is associated with

Reviewer: ```suggestion
- **user**: Username the token is associated with
```

> - **permissions**: List of permissions and resources

Reviewer: ```suggestion
- **permissions**: List of permissions for organization resources
```

**Copy the generated `token` and store it for safe keeping.**

<!------------------------------ END API CONTENT ------------------------------>
{{% /tab-content %}}
    {{< /tabs-wrapper >}}

    {{% note %}}

> We recommending using a password manager or a secret store to securely store

Reviewer: ```suggestion
We recommend using a password manager or a secret store to securely store
```

sensitive tokens.
    {{% /note %}}

5.  {{< req text="(Optional)" color="magenta" >}} **Create a bucket**

    In the InfluxDB initialization process, you created a bucket.
    You can use that bucket or create a new one specifically for this getting
    started tutorial. All examples in this tutorial assume a bucket named

>     **get-started**.

Reviewer: ```suggestion
    _get-started_.
```

3.  Click **+ {{< caps >}}Create bucket{{< /caps >}}**.
4.  Provide a bucket name (get-started) and select {{% caps %}}Never{{% /caps %}}
    to create a bucket with an infinite [retention period](/influxdb/v2.4/reference/glossary/#retention-period).
5.  Click **{{< caps >}}Create{{< /caps >}}**.

<!------------------------------- END UI CONTENT ------------------------------>
{{% /tab-content %}}
{{% tab-content %}}
<!---------------------------- BEGIN CLI CONTENT ----------------------------->

1.  If you haven't already, [download, install, and configure the `influx` CLI](/influxdb/v2.4/tools/influx-cli/).
2.  Use the [`influx bucket create` command](/influxdb/v2.4/reference/cli/influx/bucket/create/)
    to create a new bucket.
    
    **Provide the following**:

    - `-n, --name` flag with the bucket name.

    {{< cli/influx-creds-note >}}

    ```sh
    influx bucket create --name get-started
    ```

<!------------------------------ END CLI CONTENT ------------------------------>
{{% /tab-content %}}
{{% tab-content %}}
<!----------------------------- BEGIN API CONTENT ----------------------------->

> To create a new bucket using the InfluxDB HTTP API, send a request to

Reviewer: ```suggestion
To create a bucket using the InfluxDB HTTP API, send a request to
```

the InfluxDB API `/api/v2/buckets` endpoint using the `POST` request method.

{{< api-endpoint endpoint="http://localhost:8086/api/v2/buckets" method="post" >}}

Include the following with your request:

- **Headers**:

>   - **Authorization**: Token <INFLUX_TOKEN>

Reviewer: ```suggestion
  - **Authorization**: Token `INFLUX_TOKEN`
```

>   - **Content-Type**: application/json

Reviewer: ```suggestion
  - **Content-Type**: `application/json`
```

- **Request body**: JSON object with the following properties:
  - **org**: InfluxDB organization name
  - **name**: Bucket name
  - **retentionRules**: List of retention rule objects that define the bucket's retention period.
    Each retention rule object has the following properties:

>     - **type**: "expire"

Reviewer: ```suggestion
    - **type**: `"expire"`
```

    - **everySeconds**: Retention period duration in seconds.
      `0` indicates the retention period is infinite. 

```sh
export INFLUX_HOST=http://localhost:8086
export INFLUX_ORG_ID=<YOUR_INFLUXDB_ORG_ID>

> export INFLUX_TOKEN=<YOUR_INFLUXDB_API_TOKEN>

Reviewer: These are already set for OSS by the setup process, but if we're telling them to set INFLUX_TOKEN again, should we specify the all access token they just created or leave it to them to decide?

curl --request POST \
"$INFLUX_HOST/api/v2/buckets" \
  --header "Authorization: Token $INFLUX_TOKEN" \
  --header "Content-Type: application/json" \
  --data '{
    "orgID": "'"$INFLUX_ORG_ID"'",

>     "name": "get-started-test",

Reviewer: ```suggestion
    "name": "get-started",
```
